programs/drift/src/instructions/user.rs

```diff
@@ -1940,6 +1973,29 @@ pub struct Deposit<'info> {
     pub token_program: Program<'info, Token>,
 }
 
+#[derive(Accounts)]
+#[instruction(market_index: u16,)]
```

Comment:
this is wrong i think, as is it it would use the `amount` as the `market_index`, im surprised the ts test passed

i'd get wrong of the market index here and just pull it from the spot market using
`spot_market.load()?.market_index`